src/sign-verify/clu_dgst_setup.c

```diff
@@ -0,0 +1,263 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#include <wolfclu/clu_header_main.h>
+#include <wolfclu/clu_optargs.h>
+#include <wolfclu/sign-verify/clu_sign.h>
+#include <wolfclu/sign-verify/clu_verify.h>
+
+int wolfCLU_dgst_setup(int argc, char** argv)
+{
+    WOLFSSL_BIO *sigBio = NULL;
+    WOLFSSL_BIO *pubKeyBio = NULL;
+    WOLFSSL_BIO *dataBio = NULL;
+    WOLFSSL_EVP_PKEY *pkey;
+    int     ret = 0;
+    char* sig  = NULL;
+    char* data = NULL;
+    void* key  = NULL;
+    int derSz  = 0;
+    int dataSz = 0;
+    int sigSz  = 0;
+    int keySz  = 0;
+    int option;
+    int longIndex = 2;
+
+    unsigned char* der = NULL;
+    ecc_key ecc;
+    RsaKey  rsa;
+    word32 idx = 0;
+
+    enum wc_HashType      hashType = WC_HASH_TYPE_NONE;
+    enum wc_SignatureType sigType  = WC_SIGNATURE_TYPE_NONE;
+
+    opterr = 0; /* do not display unrecognized options */
+    optind = 0; /* start at indent 0 */
+    while ((option = getopt_long_only(argc, argv, "",
+                   dgst_options, &longIndex )) != -1) {
+
+        switch (option) {
+
+            case WOLFCLU_CERT_SHA:
+                hashType = WC_HASH_TYPE_SHA;
+                break;
+
+            case WOLFCLU_CERT_SHA224:
+                hashType = WC_HASH_TYPE_SHA224;
+                break;
+
+            case WOLFCLU_CERT_SHA256:
+                hashType = WC_HASH_TYPE_SHA256;
+                break;
+
+            case WOLFCLU_CERT_SHA384:
+                hashType = WC_HASH_TYPE_SHA384;
+                break;
+
+            case WOLFCLU_CERT_SHA512:
+                hashType = WC_HASH_TYPE_SHA512;
+                break;
+
+            case WOLFCLU_VERIFY:
+                pubKeyBio = wolfSSL_BIO_new_file(optarg, "rb");
+                if (pubKeyBio == NULL) {
+                    printf("unable to open public key file %s\n", optarg);
+                    ret = -1;
+                }
+                break;
+
+            case WOLFCLU_INFILE:
+                sigBio = wolfSSL_BIO_new_file(optarg, "rb");
+                if (sigBio == NULL) {
+                    printf("unable to signature file %s\n", optarg);
+                    ret = -1;
+                }
+                break;
+
+            case ':':
+            case '?':
+                break;
+
+            default:
+                /* do nothing. */
+                (void)ret;
+        }
+    }
+
+    /* signed file should be the last arg */
+    if (ret == 0) {
+        dataBio = wolfSSL_BIO_new_file(argv[argc-1], "rb");
+        if (dataBio == NULL) {
+            printf("unable to open data file %s\n", argv[argc-1]);
```

Comment:
"data file" isn't very clear. Maybe "signed file", like it's called in the comment at 105?